xbmc/video/Bookmark.h

```diff
@@ -38,20 +38,24 @@ class CBookmark
    */
   bool IsPartWay() const;
 
+  bool IsMediaBookmark() const;
+  static bool IsMediaBookmark(int t);
+
   double timeInSeconds;
   double totalTimeInSeconds;
   long partNumber;
   std::string thumbNailImage;
   std::string playerState;
   std::string player;
-  long seasonNumber;
-  long episodeNumber;
+  std::string title;
 
   enum EType
   {
     STANDARD = 0,
     RESUME = 1,
-    EPISODE = 2
+    EPISODE = 2,
```

Comment:
We have tried to make new things as generic as possible. That's why most link tables have been moved from one link table per media type to a single link table with a `media_type` column. So personally I'd prefer to have it generic here as well i.e. use `MEDIA` here, store the media type in the bookmark table if necessary and pass it as an extra parameter to the necessary methods.

Yes if and when a new media type is added it will need extensions in a lot of places but the fewer places there are the easier it will be and the less bugs will appear.